assets/templates/aws.sqs/sqs.gotext.log

```diff
@@ -0,0 +1,2 @@
+{ "create" : { "_index": "metrics-aws.sqs-default" } }
+{ "@timestamp": "2022-07-26T21:43:00.000Z", "agent": { "name": "docker-fleet-agent", "id": "2d4b09d0-cdb6-445e-ac3f-6415f87b9864", "type": "metricbeat", "ephemeral_id": "cdaaaabb-be7e-432f-816b-bda019fd7c15", "version": "8.3.2" }, "elastic_agent": { "id": "2d4b09d0-cdb6-445e-ac3f-6415f87b9864", "version": "8.3.2", "snapshot": false }, "cloud": { "provider": "aws", "region": "{{ generate "Region" }}", "account": { "name": "elastic-beats", "id": "000000000000" } }, "ecs": { "version": "8.0.0" }, "service": { "type": "aws" }, "data_stream": { "namespace": "default", "type": "metrics", "dataset": "aws.sqs" }, "metricset": { "period": 300000, "name": "cloudwatch" }, "event": { "duration": {{ generate "EventDuration" }}, "agent_id_status": "verified", {{ $eventIngested := generate "EventIngested" }} "ingested": "{{ $eventIngested.Format "2006-01-02T15:04:05.999999Z07:00" }}", "module": "aws", "dataset": "aws.sqs" }, "aws": { "cloudwatch": { "namespace": "AWS/SQS" }, "dimensions": { "QueueName": "{{ generate "QueueName" }}" }, "sqs": { "metrics": { "ApproximateAgeOfOldestMessage": { "avg": {{ generate "OldestMessageAge" }} }, "ApproximateNumberOfMessagesDelayed": { "avg": {{ generate "Delayed" }} }, "ApproximateNumberOfMessagesNotVisible": { "avg": {{ generate "NotVisible" }} }, "ApproximateNumberOfMessagesVisible": { "avg": {{ generate "Visible" }} }, "NumberOfMessagesDeleted": { "avg": {{ generate "Deleted" }} }, "NumberOfMessagesReceived": { "avg": {{ generate "Received" }} }, "NumberOfMessagesSent": { "avg": {{ generate "Sent" }} }, "NumberOfEmptyReceives": { "avg": {{ generate "EmptyReceives" }} }, "SentMessageSize": { "avg": {{ generate "SentMessageSize" }} }, } }, "tags": { "createdBy": {{ generate "TagsCreatedBy" }}, } }, "cloud": { "account": { "id": "000000000000", "name": "elastic-observability" }, "provider": "aws", "region": {{ generate "Region" }} } }
```

Comment:
I wonder if there is a way that we could allow users to build is template in "pretty json" to make it much more readable, simplify PR reviews etc. and only on generation munch it into a signle line?

Reply:
I see the value of building a "pretty json" template, but we have to consider that a template can be something else than a json, and having multiple lines can be semantically meaningful in a different context.

We have to provide a flag somewhere (when running the command/in some config file/etc) if a specific template must be "folded" in a single line or not, if we want to provide such option

Reply:
yes. My thinking is that "somewhere" there will be config for all the generation task and the templates linked to it. One way to indicate it could also be the file ending.